engine/include/components/rigidbody_component.hpp

```diff
@@ -1,19 +1,47 @@
 ﻿#pragma once
-#include "modules/physics_module.hpp"
+#include "physics_module.hpp"
 
 struct RigidbodyComponent
 {
     // default creates a sphere at 0,2,0
-    RigidbodyComponent(JPH::BodyInterface& bodyInterface)
+    RigidbodyComponent(JPH::BodyInterface& bodyInterface, PhysicsShapes shape = eSPHERE, entt::entity ownerEntity = entt::null)
     {
-        JPH::BodyCreationSettings sphere_settings(new JPH::SphereShape(0.5f), JPH::Vec3(0.0, 2.0, 0.0), JPH::Quat::sIdentity(), JPH::EMotionType::Dynamic, PhysicsLayers::MOVING);
-        bodyID = bodyInterface.CreateAndAddBody(sphere_settings, JPH::EActivation::Activate);
+        shapeType = shape;
+        if (shape == eSPHERE)
+        {
+            JPH::BodyCreationSettings const bodySettings(new JPH::SphereShape(0.5f), JPH::Vec3(0.0, 2.0, 0.0), JPH::Quat::sIdentity(), JPH::EMotionType::Dynamic, PhysicsLayers::MOVING);
+            bodyID = bodyInterface.CreateAndAddBody(bodySettings, JPH::EActivation::Activate);
+        }
+        else if (shape == eBOX)
+        {
+            JPH::BodyCreationSettings const bodySettings(new JPH::BoxShape(JPH::Vec3(0.5, 0.5, 0.5)), JPH::Vec3(0.0, 2.0, 0.0), JPH::Quat::sIdentity(), JPH::EMotionType::Dynamic, PhysicsLayers::MOVING);
+            bodyID = bodyInterface.CreateAndAddBody(bodySettings, JPH::EActivation::Activate);
```

Comment:
doesnt cover eCONVEX_HULL case, if we don't support this should remove it from the enum or throw error/message